Add missing X-Pack repositories REST specs

stevejgordon opened the pull request:
While working on the .NET client code-gen, I noticed that a new experimental X-Pack API was added for repositories in 7.10.0, but is missing a REST spec required for client code generation. This PR adds the specs for the two new endpoints.

API was added in #60371

Please let me know what (if anything) needs to happen to backport to 7.10 and 7.x?

elasticmachine commented:
Pinging @elastic/clients-team (Team:Clients)

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

Mpdreamz commented on x-pack/plugin/src/test/resources/rest-api-spec/api/repositories.clear_metering_archive.json:
```diff
@@ -0,0 +1,30 @@
+    "url":{
+      "paths":[
+        {
+          "path":"/_nodes/{node_id}/_repositories_metering/{max_version_to_clear}",
```
```suggestion
          "path":"/_nodes/{node_id}/_repositories_metering/{version}",
```

fcofdez replied:
I'll prefer to leave that parameter name, as it clears up to the specified version.

Mpdreamz reviewed (approved):
LGTM

Mpdreamz commented:
@fcofdez this LGTM, mind giving this a quick look over since this is your new API? 👍

Mpdreamz commented:
Jenkins run elasticsearch-ci/2 please

fcofdez reviewed (approved):
LGTM

stevejgordon committed:
Add missing repositories API REST specs

A new experimental X-Pack API was added for repositories, but is missing a REST spec. The API was added in #60371

stevejgordon commented:
Apologies @fcofdez, I've just noticed that I let this PR get stale. Since these still need to be added I have rebased them and updated them to include some of the latest spec fields such as `visibility`. Which version labels would be best to add these? Would 7.15.0 and 8.0.0 be enough?

stevejgordon committed:
Move this under the nodes namespace

stevejgordon committed:
Rename max_version_to_clear parameter

stevejgordon committed:
Update docs URL to use current

fcofdez commented:
No worries @stevejgordon :). `7.15.0` and `8.0.0` versions are fine. 👍

elasticsearchmachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/elasticsearch/pull/75412)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/75412">](https://github.com/elastic/elasticsearch/pull/75412) |
